data/balance/coin.go

```diff
@@ -114,24 +116,27 @@ func (coin Coin) Equals(value Coin) bool {
 }
 
 // Minus two coins
-func (coin Coin) Minus(value Coin) Coin {
+func (coin Coin) Minus(value Coin) (Coin, error) {
```

Comment:
if you change the Minus return type, probably also change the Plus, and so on

Reply:
fixed this